zstacklib/zstacklib/utils/linux.py

```
@@ -744,13 +753,24 @@ def get_percentage(filesize, dst):
             if 'Content-Length' in l:
                 filesize = l.split(':')[1].strip()
                 return True, int(filesize)
         return False, 0
 
+    def is_cancelled():
+        return cancellation_checker and cancellation_checker()
+
+    def kill_download_process(process):
+        try:
+            kill_all_child_process(process.pid)
+        except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)